graphdatascience/tests/unit/test_aurads_sessions.py

```diff
@@ -457,29 +431,5 @@ def test_estimate_size_exceeds(aura_api: AuraApi) -> None:
         assert sessions.estimate(1, 1, [AlgorithmCategory.CENTRALITY]) == SessionMemory.m_8GB
 
 
-def test_from_specific_instance_details() -> None:
-    info = SessionInfo.from_specific_instance_details(
-        InstanceSpecificDetails("id", "gds-session-foo", "tenant", "cp", "status", "bolt", "96GB", "type", "region")
-    )
-
-    assert info.name == "foo"
-    assert info.memory == "96GB"
-
-
-def test_from_specific_instance_details_failures() -> None:
-    with pytest.raises(ValueError, match="Unknown memory configuration: `invalid`. Supported values are `\\['8GB', "):
-        SessionInfo.from_specific_instance_details(
-            InstanceSpecificDetails(
-                "id", "gds-session-foo", "tenant", "cp", "status", "bolt", "invalid", "type", "region"
-            )
-        )
-    with pytest.raises(ValueError, match="Invalid session name: `not-a-gds-session-foo`"):
-        SessionInfo.from_specific_instance_details(
-            InstanceSpecificDetails(
-                "id", "not-a-gds-session-foo", "tenant", "cp", "status", "bolt", "24GB", "type", "region"
-            )
-        )
-
-
```

Comment:
removing these tests because the method tested is removed -> fine

the replacement method is `SessionDetails.fromJson`, which is untested. We should add back these tests for that method. I suppose that ends up solving the TODO about parsing session memory as well.

Reply:
I realized, we dont need to parse the memory field, as in the end we endup using the str version and not the enum.
Imagine for having a better str representation?

Reply:
You're right we don't need to, but I think it's nice if we do it anyway, for general type safety.

Reply:
How will we add new memory options to the sessions?

First we add them to Aura, but then (old) clients will complain about the new unknown size.
We would first need to add them to the client and get everyone to upgrade.

IMO thats not a good UX. Thats why I would rather keep the str format and only offer the typed size on the API calls. This makes it easier for users to chose which size they can select, but not throw on new sizes

Reply:
That makes a lot of sense. Yes, better to not parse into the enum!